extension/src/ALObject/ALParser.ts

```diff
@@ -109,52 +109,50 @@ function parseProcedureDeclaration(
     const attributes: string[] = [];
     let lineNo = procedureLineNo - 1;
     let loop = true;
-    do {
-      const line = alCodeLines[lineNo].code;
-      const attributeMatch = line.match(attributePattern);
-      if (attributeMatch) {
-        if (attributeMatch.groups?.attribute) {
-          attributes.push(attributeMatch[0].trim());
+    if (lineNo > 0) {
+      do {
```

Comment:
Maybe replace with a while-do? (`while (lineNo >= 0) { ... }`). Would also apply for the next one on L133.

Reply:
Nope... There's another condition that changes `loop` as well.

Reply:
Sry. Me being blind as usual :D